agent/xds/rbac.go

```diff
@@ -580,37 +634,52 @@ func idPrincipal(src structs.ServiceName) *envoy_rbac_v3.Principal {
 	}
 }
 
-func makeSpiffePattern(sourceAP, sourceNS, sourceName string) string {
-	if sourceNS == structs.WildcardSpecifier && sourceName != structs.WildcardSpecifier {
-		panic(fmt.Sprintf("not possible to have a wildcarded namespace %q but an exact service %q", sourceNS, sourceName))
+const anyPath = `[^/]+`
```

Comment:
At one point in my life I preferred grouping them at the top of a file but dnephin convinced me they are better left closer to their usages. No strong preference but since this isn't exported I think there's little value moving it to the top

Reply:
Another way to think of it is that "business logic" constants might make more sense at the top of a file, for visibility into arbitrary knobs we might want to tweak.

Plumbing constants like this one don't need visibility and thus can live closer to where they are used.

Reply:
Yeah if we're writing libraries with consts / enums / sentinel errors that users should know and/or easily discover, those are best grouped at the top of files (alternatively I've seen them defined in consts.go)